Samples/AudioCreation/cs/AudioCreation/Scenario7_RecordPlayback.xaml.cs

```diff
@@ -0,0 +1,570 @@
+                        capacityInBytes = (uint)(graph.LatencyInSamples << 3);
+                    }
+                }
+
+                // if we fill up, just spin forever.  don't try to exit recording from audio thread,
+                // we don't bother with two-way signaling.
+                if (maxIndex + capacityInBytes > byteBuffer.Length)
+                {
+                    Debug.Assert(byteBuffer.Length >= maxIndex);
+                    capacityInBytes = (uint)byteBuffer.Length - maxIndex;
+                }
+
+                fixed (byte* buf = byteBuffer)
+                {
+                    for (int i = 0; i < capacityInBytes; i++)
+                    {
+                        byte b = dataInBytes[i + bufferStart];
+                        buf[i + maxIndex] = b;
+
+                        if (b != 0)
+                        {
+                            nonZeroSampleCount++;
+                        }
+                    }
+                }
+                maxIndex += capacityInBytes;
+            }
+
+            audioFrameCount++;
+        }
+
+        private void UpdateStatusWhileRecording()
+        {
+            uint max = maxIndex >> 3;
+            uint curr = currentIndex >> 3;
+            int frameCount = audioFrameCount;
+            // may have stopped recording while this async notification was in flight
+            if (isRecording)
+            {
+                rootPage.NotifyUser($"recording: max samples {max}, frame count {frameCount}, last cap {lastCapacityEncountered}, max cap {maxCapacityEncountered}", NotifyType.StatusMessage);
+            }
+        }
+
+        private void FrameInputNode_QuantumStarted(AudioFrameInputNode sender, FrameInputNodeQuantumStartedEventArgs args)
+        {
+            uint requiredSamples = (uint)args.RequiredSamples;
+
+            if (requestStartPlaying)
+            {
+                Debug.Assert(!isRecording);
+                Debug.Assert(!requestStartRecording);
+                Debug.Assert(!isPlaying);
+
+                isPlaying = true;
+                requestStartPlaying = false;
+                currentIndex = 0;
+            }
+
+            if (isPlaying)
+            {
+                Debug.Assert(!isRecording);
+                Debug.Assert(!requestStartPlaying);
+                Debug.Assert(!requestStartRecording);
+
+                HandleOutgoingAudio(requiredSamples);
+
+                /*
+                int currentStatusMsec = DateTime.Now.Millisecond;
+                // update status every tenth of a second
+                if (lastStatusMsec % 100 > currentStatusMsec % 100)
+                {
+                    CoreApplication.MainView.CoreWindow.Dispatcher.TryRunAsync(
+                        Windows.UI.Core.CoreDispatcherPriority.Normal,
+                        UpdateStatusWhilePlaying);
+                }
+                lastStatusMsec = currentStatusMsec;
+                */
```

Comment:
Opted to add comment; it's an OK example of cross-thread UI update during audio processing.